### PR TITLE
Remove some of the magic from emcmake

### PR DESCRIPTION
**Pull request by sbc100**
emcmake should be a tools that simply wraps cmake.  We should not be
opinionated about the generate that a given user might want to use of
try to work around MinGW issues on behalf of the user.

### CODE REVIEW

**Review by dschuff — APPROVED** [review]
LGTM in general (in the sense that I agree emscripten doesn't need this behavior). Since this will break users who rely on the default, it's probably also worth a post to the mailing list in addition to the release notes.

**Comment by juj** [comment]
Noticed today that emcmake support has regressed after updating to latest. The reason for this code was that on Windows CMake defaults to targeting Visual Studio if no generator is otherwise present, but Visual Studio+Emscripten combo does not work without a custom plugin (like juj/vs-tool was).

After this landed, the default form of

```
cd /path/to/emscripten/project/
emcmake cmake .
cmake --build .
```

is now broken on Windows, and will lead to random compilation errors as cmake tries to shove all the Emscripten code through Visual Studio instead.

The idea was to default to using MinGW Makefiles instead to avoid this issue - mingw32-make is shipped via emsdk.

The reason why sh.exe was removed from PATH is that if one has Git installed and set up to PATH, it would come with its own copy of sh.exe which breaks mingw32-make.exe.

So basically instead of being able to do the above on Windows, one now has to

```
cd /path/to/emscripten/project/
set ORIGPATH=%PATH%
set PATH=%PATH:Git=Git_unused%
emcmake cmake . -G "MinGW Makefiles"
set PATH=%ORIGPATH%
cmake --build .
```

to work around these issues, which is quite awkward. I can appreciate the desire for simplicity in the codebase, but have to say that it is quite a frustrating experience to find about four regressions in Emscripten on Windows when I updated to latest version last week, and to find out the root cause was akin to a "spring cleaning". :/

**Comment by sbc100** [comment]
Sorry, I was not aware of the "Visual Studio+Emscripten combo does not work" issue.  I also had no idea that mingw32-make was shipped with emsdk.

We can revert this change if needed...

Do you know why the Visual Studio generator doesn't just work?  It seems like it should work fine since cmake is supposed to be agnostic to the generator used, even if the compiler is overridden (at least I thought that was how cmake was supposed to work but maybe I misunderstood.).  Does the "Visual Studio" generator not work with random values of `CMAKE_CXX_COMPILER=xxx`?

**Comment by sbc100** [comment]
Will revert for now since I don't have easy access to a windows machine .. its seems like I misjudged the usefulness of this feature.

@aardappel I guess I was wrong about emscripten + visual studio being thing that just works.

**Comment by aardappel** [comment]
in an ideal world without leaky abstractions :)

**Comment by juj** [comment]
> Do you know why the Visual Studio generator doesn't just work? It seems like it should work fine since cmake is supposed to be agnostic to the generator used, even if the compiler is overridden (at least I thought that was how cmake was supposed to work but maybe I misunderstood.). Does the "Visual Studio" generator not work with random values of `CMAKE_CXX_COMPILER=xxx`?

CMake tries to pass Visual Studio compiler arguments to Emscripten then. Even when the path to the compiler is changed, it will still think it is a path to a Visual Studio cl.exe.
